### PR TITLE
[messages,flags] Shrinkify vcpkg-tool

### PR DESCRIPTION
**Pull request by strega-nil-ms**
I was bored on the bus, so I decided to play around with making the vcpkg binary smaller. Tested on an ARM64 macbook:

```
# Windows ARM64
6'241'792 vcpkg.exe       # -0
6'203'904 sc-vcpkg.exe    # -38K
5'200'384 os-vcpkg.exe    # -1.04M
5'170'688 os-sc-vcpkg.exe # -1.07M

# macOS ARM64
2'695'538 vcpkg       # -0
2'675'234 sc-vcpkg    # -20K
2'669'826 os-vcpkg    # -26K
2'645'730 os-sc-vcpkg # -50K
```

Both together result in a binary size reduction of 17% on Windows, and 1.8% on macOS (obviously not as big of a deal, but still a 50K reduction)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -111,10 +111,10 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
 if(MSVC)
     string(APPEND CMAKE_CXX_FLAGS " /EHsc")
-    if(CMAKE_BUILD_TYPE STREQUAL "Release")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi /guard:cf")
-        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG /debugtype:cv,fixup /guard:cf")
-    endif()
+    string(APPEND CMAKE_CXX_FLAGS_RELEASE " /Zi /guard:cf /Os")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS_RELEASE " /DEBUG /debugtype:cv,fixup /guard:cf")
+else()
+    string(REGEX REPLACE "-O[0-3]" "-Os" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
```
Comment (dg0yt) [inline CMakeLists.txt:117]: If you want `MinSizeRel`, why do you build `Release`?

Comment (strega-nil-ms) [reply CMakeLists.txt:117]: fair point, I forgot about MinSizeRel

Comment (ras0219-msft) [reply CMakeLists.txt:117]: We should probably suck out all these flag configurations into an "Official Build Toolchain"

**Comment by BillyONeal** [comment]
CI is failing (1) because of a clang-format needed and (2) because this change causes x-generate-default-message-map to crash.

image

**Comment by BillyONeal** [comment]
@strega-nil @strega-nil-ms  Are you still working on this one?

**Comment by strega-nil-ms** [comment]
Closing as I am no longer on the team.

**Pull request closed without merging**